lib/modules/manager/mise/extract.ts

```diff
@@ -0,0 +1,109 @@
+import is from '@sindresorhus/is';
+import { logger } from '../../../logger';
+import type { ToolingConfig } from '../asdf/upgradeable-tooling';
+import type { PackageDependency, PackageFileContent } from '../types';
+import type { MiseToolSchema } from './schema';
+import {
+  ToolingDefinition,
+  asdfTooling,
+  miseTooling,
+} from './upgradeable-tooling';
+import { parseTomlFile } from './utils';
+
+export function extractPackageFile(
+  content: string,
+  packageFile: string,
+): PackageFileContent | null {
+  logger.trace(`mise.extractPackageFile(${packageFile})`);
+
+  const misefile = parseTomlFile(content, packageFile);
+  if (!misefile) {
+    return null;
+  }
+
+  const deps: PackageDependency[] = [];
+  const tools = misefile.tools;
+
+  if (tools) {
+    for (const [name, toolData] of Object.entries(tools)) {
+      const version = parseVersion(toolData);
+      const depName = name.trim();
+      const toolConfig = getToolConfig(depName, version);
+      const dep = createDependency(depName, version, toolConfig);
+      deps.push(dep);
+    }
+  }
+
+  return deps.length ? { deps } : null;
+}
+
+function parseVersion(toolData: MiseToolSchema): string | null {
+  if (is.nonEmptyString(toolData)) {
+    // Handle the string case
+    // e.g. 'erlang = "23.3"'
+    return toolData;
+  }
+  if (is.array(toolData, is.string)) {
+    // Handle the array case
+    // e.g. 'erlang = ["23.3", "24.0"]'
+    return toolData[0]; // Get the first version in the array
+  }
+  if (is.object(toolData)) {
+    // Handle the object case with a string version
+    // e.g. 'python = { version = "3.11.2" }'
+    if (is.string(toolData.version)) {
+      return toolData.version;
+    }
+  }
```

Comment:
```suggestion
  if (is.nonEmptyString(toolData.version)) {
    // Handle the object case with a string version
    // e.g. 'python = { version = "3.11.2" }'
    return toolData.version;
  }
```
Because of type narrowing we already know that it is a object and based on your implementation this should return null for empty string